### DISCUSSION TITLE
I am not sure how to log in to Functor Network, can you help me out, please?

### DISCUSSION THREAD

**Discussion by nepomnyi** [c-ding-math/functor-network#17] [Q&A] [answered]
Hello, dear Functor Network community.
I registered on the Functor website yesterday, verified my email and logged out. Then I tried to log in back and failed. I ended up resetting the password in order to log in. All of that happened yesterday. Just now, I have tried to log in again and failed again. To log in, I am using my email and the password saved in my Firefox passwords. I.e., Firefox just autofills the user name and password for me.
I am wondering what I am doing wrong.
Thank you.
Ivan

**Comment by c-ding-math** [comment]
> Hello, dear Functor Network community. I registered on the Functor website yesterday, verified my email and logged out. Then I tried to log in back and failed. I ended up resetting the password in order to log in. All of that happened yesterday. Just now, I have tried to log in again and failed again. To log in, I am using my email and the password saved in my Firefox passwords. I.e., Firefox just autofills the user name and password for me. I am wondering what I am doing wrong. Thank you. Ivan

If you can log in with the autofilled password, then you should be able to log in by hand. You may find out the password that your Firefox saves for you and check whether it coincides with what you set.

**Comment by nepomnyi** [comment, marked as answer]
I have just tried it (typing the password by hand) and it failed again. 
But I figured out the issue eventually. My email has capital letters in it. However, I noticed that Functor Network has made all the letters in my email small. So I went ahead and typed my email with small letters on the log-in page. And I was logged in! So, my issue is resolved.
I know people usually say that it doesn't matter if the email has capital letters in it because no log-in client cares about it. However, I do remember I already had problem with this once, several years ago. I don't remember what website it was, though. And now I am having this issue with Functor.
I am wondering if you think this is a bug.
Thanks.
Ivan

**Reply by c-ding-math** [reply]
Your email is normalized on registration, but not on login. It's a bug, and should be fixed in next version.

**Reply by c-ding-math** [reply]
> I have just tried it (typing the password by hand) and it failed again. But I figured out the issue eventually. My email has capital letters in it. However, I noticed that Functor Network has made all the letters in my email small. So I went ahead and typed my email with small letters on the log-in page. And I was logged in! So, my issue is resolved. I know people usually say that it doesn't matter if the email has capital letters in it because no log-in client cares about it. However, I do remember I already had problem with this once, several years ago. I don't remember what website it was, though. And now I am having this issue with Functor. I am wondering if you think this is a bug. Thanks. Ivan

The email address standard RFC5322 does not require case insensitivity. The case sensitivity, as well as some other additional rules, is totally dependent on email providers, and there is an email provider I know that treats A@x.com and a@x.com differently. So, we only filter the email address according to the standard  and we won't lowercase user's email address on registration. 

For your case, you can log in to your account with the lowercase email, add your favorite email address and delete the lowercase one.

(But one can register two accounts with A@gamil.com and a@gmail.com? We simply allow this)
